hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -124,12 +126,17 @@ public class HoodieWriteConfig extends HoodieConfig {
       .withDocumentation("Payload class used. Override this, if you like to roll your own merge logic, when upserting/inserting. "
           + "This will render any value set for PRECOMBINE_FIELD_OPT_VAL in-effective");
 
-  public static final ConfigProperty<String> MERGE_CLASS_NAME = ConfigProperty
+  public static final ConfigProperty<String> MERGER_CLASS_NAME = ConfigProperty
```

Comment:
I wonder if we can simplify all this by having a table property that takes a list of merger class names.  Then we can call `getRecordType()` to pick the optimal merge implementation? But high level should n't this be a table config like before.

Reply:
I have discussed this with Raymond. RecordMerger should not be persisted in table config. It is up to the user to specify what to choose. If not specified, we provide a engine-default RecordMerger.

Reply:
We probably fine either way, but for simplicity i'd suggest we stick with our current approach of keeping it table-property:

1. It's easier for users to understand (existing behavior)
2. It stays part of the table's (immutable) configuration, avoiding the toll of needing to specify it explicitly with every write (could be error-prone)

Reply:
@vinothchandar You left comments here. "But high level should n't this be a table config like before."